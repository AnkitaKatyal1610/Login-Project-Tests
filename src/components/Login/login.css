.Login {
    margin: auto;
    width: 300px;
    margin-top: 75px;
    text-align: center;
    border: 3px solid grey;
    padding: 15px;
    border-radius: 9px 
}