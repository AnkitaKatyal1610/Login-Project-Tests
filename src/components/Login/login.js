import React, { Component } from 'react';
import { Input, Button, Form, Label } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import { withRouter } from 'react-router-dom';
import { loginAction } from '../../action/action'
import './login.css'

export class Login extends Component {

    state = {
        email: '',
        password: '',
        submitted: false
    }
    handleSubmit = () => {
        this.setState({ submitted: true })
        let { email, password } = this.state;
        if (email !== '' && password !== '') {
            this.props.loginAction({ email, password }).then(() => {
                if (localStorage.getItem('token'))
                    this.props.history.replace('/homepage');
            });
        }
    }

    render() {
        return (
            <div className='Login'>
                <Form>
                    <Form.Field>
                        <Input value={this.state.email} icon='user circle' iconPosition='left' type='email' onChange={(event) => { this.setState({ email: event.target.value, submitted: false, invalid: false }) }} placeholder='Email' />
                        {this.state.submitted && this.state.email === '' ? <Label color='red' pointing>Please enter your email</Label> : null}
                    </Form.Field>
                    <Form.Field>
                        <Input value={this.state.password} icon='lock' iconPosition='left' type='password' onChange={(event) => { this.setState({ password: event.target.value, submitted: false }) }} placeholder='Password' />
                        {this.state.submitted && this.state.password === '' && this.state.email !== '' ? <Label color='red' pointing>Please enter your password</Label> : null}
                    </Form.Field>
                    <Button onClick={this.handleSubmit} color='blue'>Login</Button>
                </Form>
            </div>
        )
    }
}

const mapDispatch = (dispatch) => {
    return {
        loginAction: bindActionCreators(loginAction, dispatch)
    }
}
export default withRouter(connect(null, mapDispatch)(Login));