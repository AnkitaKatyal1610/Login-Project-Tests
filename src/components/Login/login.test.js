import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Login } from './login';

Enzyme.configure({ adapter: new Adapter() });

let promise = new Promise((resolve, reject) => {
    setTimeout(() => {
        resolve();
    }, 500);
});

describe('login component', () => {
    let props = {
        loginAction: jest.fn(() => promise)

    };
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Login {...props} />);
    })
    it('has two text inputs', () => {
        const textInputs = wrapper.find('Input')
        expect(textInputs.length).toEqual(2);
    })

    it('has a login button', () => {
        const buttons = wrapper.find('Button');
        expect(buttons.length).toEqual(1);
        expect(buttons.prop('children')).toBe('Login');
    })

    it('has initial state empty', () => {
        const { email, password } = wrapper.state();
        expect(email).toBe("");
        expect(password).toBe("");
    })

    it('changes the state on change event in the input fields', () => {
        const emailInput = wrapper.find('Input[type="email"]')
        const passwrdInput = wrapper.find('Input[type="password"]')
        emailInput.simulate('change', {
            target: { value: 'test@test.com' }
        })
        passwrdInput.simulate('change', {
            target: { value: 'test' }
        })
        const { email, password } = wrapper.state();
        expect(email).toBe("test@test.com");
        expect(password).toBe("test");
    })

    it('calls onClick event handler when Login button is clicked', () => {
        let spy = jest.spyOn(wrapper.instance(), 'handleSubmit')
        wrapper.update();
        wrapper.instance().forceUpdate();
        const button = wrapper.find('Button');
        button.simulate('click');
        expect(spy).toHaveBeenCalled();
    })
    it('does not call login action if email and password are not given', () => {
        expect(wrapper.state().email).toBe('')
        expect(wrapper.state().password).toBe('')
        const button = wrapper.find('Button');
        button.simulate('click');
        expect(props.loginAction).toHaveBeenCalledTimes(0)
    })
    it('calls login action when email and password are set', () => {
        const emailInput = wrapper.find('Input[type="email"]')
        const passwrdInput = wrapper.find('Input[type="password"]')
        emailInput.simulate('change', {
            target: { value: 'test@test.com' }
        })
        passwrdInput.simulate('change', {
            target: { value: 'test' }
        })
        const button = wrapper.find('Button');
        button.simulate('click');
        expect(props.loginAction).toHaveBeenCalled()
    })


})
