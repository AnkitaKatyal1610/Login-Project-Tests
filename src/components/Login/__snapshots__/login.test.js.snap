// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component matches snapshot 1`] = `
<div
  className="Login"
>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <input
          onChange={[Function]}
          placeholder="Email or Username"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="user circle icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <input
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <i
          aria-hidden="true"
          className="lock icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <button
      className="ui blue button"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
</div>
`;
